runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectMetrics.java

```diff
@@ -275,13 +275,40 @@ private boolean matches(MetricsFilter filter, MetricKey key) {
         && matchesScope(key.stepName(), filter.steps());
   }
 
-  private boolean matchesScope(String actualScope, Set<String> scopes) {
+  /**
+  * {@code subPathMatches(actualScope, filterScope)} returns true if {@code filterScope}
+  * represents a path within {@code actualScope}. For example, "foo/bar" is in "a/foo/bar/b",
+  * but not "a/fool/bar/b" or "a/foo/bart/b".
+  */
+  public boolean subPathMatches(String actualScope, String filterScope) {
```

Comment:
Either change the parameter names to `needle` and `haystack` to match the comments below, or change the comments below. I'd suggest those names (or similar) to make it really clear which one is being searched for and which is being searched in.